app/scripts/controllers/transactions/index.js

```diff
@@ -193,12 +193,12 @@ class TransactionController extends EventEmitter {
     }
     txUtils.validateTxParams(normalizedTxParams)
     // construct txMeta
```

Comment:
Nit: maybe a comment here would be useful, to explain that this step _must_ happen synchronously, before anything else, because it assigns the tx id? It's really not obvious what's going on here at first glance.

This is a follow up from this discussion: https://github.com/MetaMask/metamask-extension/pull/7473#discussion_r348539205

Reply:
47bd579c8